engine/src/execution_graph/logic_controllers/BatchProcessing.cpp

```diff
@@ -311,8 +337,12 @@ kstatus BindableTableScan::run() {
 
         std::unique_lock<std::mutex> lock(kernel_mutex);
         kernel_cv.wait(lock,[this]{
-            return this->tasks.empty();
+            return this->tasks.empty() || ral::execution::executor::get_instance()->has_exception();
```

Comment:
@rommelDB I am guessing that this is the prototype of how you want to handle the exceptions? and then you would add it to all the other kernels? If so, this PR should be WIP not in review

Reply:
I am wondering how this logic would behave if multiple exceptions occur.  We could end up only serving up one of the several exceptions. And additionally all the kernels would throw the same exception. How does this look to the user? DOes the user end up with multiple repeated exception messages or just one? @rommelDB could you please test out what would happen if multiple kernels throw an error at the same time

Reply:
@williamBlazing Regarding the multiple exceptions, which would be the ideal output? at this time, when two kernels throw an exception, both error messages are shown to the user. Even if there are the same kernel or different ones.